tests/rptest/scale_tests/extreme_recovery_test.py

```diff
@@ -0,0 +1,208 @@
+
+    def validate_cluster(self, baseline, restored):
+        """Validate all data after recovery."""
+        self.logger.info(
+            f"vallidate_cluster base:restored: {baseline}:{restored}")
+
+        self._validate_partition_last_offset()
+        expected_topics = [
+            topic.name for topic in self.expected_recovered_topics
+        ]
+
+        # consume and validate all data
+        t0 = time()
+        consumers: list[FranzGoVerifiableSeqConsumer] = list()
+        for topic in self.topics:
+            self.logger.info(
+                f'validate_cluster: starting consumer for "{topic}"')
+            c = FranzGoVerifiableSeqConsumer(self.ctx, self.redpanda, topic,
+                                             self.MSG_SIZE)
+            c.start()
+
+        for c in consumers:
+            self.logger.info(
+                f'validate_cluster: waiting on consumer for "{c._topic}"')
+            c.shutdown()
+            c.wait()
+            assert c.consumer_status.invalid_reads == 0, "No invalid reads expected."
+            assert c.consumer_status.valid_reads == self._msg_per_topic
+
+        delta_t = time() - t0
+        self.logger.info(
+            f'validate_cluster: consumed {self._total_bytes/2**20}MB in {delta_t} sec;'
+            f' {(self._total_bytes/delta_t)/2**20} MB/s.')
+
+    @property
+    def second_restart_needed(self):
+        """Return True if the restart is needed afer first validation steps"""
+        # XXX TODO
+        return False
+
+    def after_restart_validation(self):
+        """Check that topic is writable after restart"""
+        # XXX TODO
+
+
+# "It's like, totally extreme, bro."
+# Try 1: hack topic recovery test into being a scale test.
+class ExtremeRecoveryTest(TopicRecoveryTest):
+    def __init__(self, test_context: TestContext):
+        extra_rp_conf = {"blah": "bloo"}
+        super(ExtremeRecoveryTest,
+              self).__init__(test_context,
+                             num_brokers=RecoveryScale.NODE_COUNT)
+        self.initial_cleanup_needed = True
+        self.ctx = test_context
+
+    def tearDown(self):
+        super(ExtremeRecoveryTest, self).tearDown()
+
+    @cluster(num_nodes=9, log_allow_list=TRANSIENT_ERRORS)
+    def test_recovery_scale(self):
+        # This test requires dedicated system resources
+        assert self.redpanda.dedicated_nodes
+
+        assert self.s3_client
+        topics = [
+            TopicSpec(partition_count=RecoveryScale.part_per_topic,
+                      replication_factor=3)
+            for i in range(RecoveryScale.NUM_TOPICS)
+        ]
+
+        extremeRecovery = RecoveryScale(self.test_context, self.s3_client,
+                                        self.kafka_tools, self.rpk,
+                                        self.s3_bucket, self.logger, topics,
+                                        self.rpk_producer_maker)
+        self.do_run(extremeRecovery, int(RecoveryScale.expected_transfer_sec))
```

Comment:
nit: i'd cast when defining the constant

Reply:
You mean type annotation on the local var? Yeah, I probably would except the right hand side is well-typed.

Reply:
I'll include this in my next update, resolving.